app/src/main/java/com/android/solvit/provider/ui/calendar/components/preferences/ScrollableTimePickerRow.kt

```diff
@@ -0,0 +1,141 @@
+package com.android.solvit.provider.ui.calendar.components.preferences
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import java.time.LocalTime
+import kotlinx.coroutines.launch
+
+@Composable
+fun ScrollableTimePickerRow(
+    selectedTime: LocalTime,
+    onTimeSelected: (LocalTime) -> Unit,
+    modifier: Modifier = Modifier,
+    testTagPrefix: String = ""
+) {
+  var activeColumn by remember { mutableStateOf<Int?>(null) }
+
+  Row(
+      modifier =
+          modifier
+              .fillMaxWidth()
+              .height(120.dp)
+              .padding(horizontal = 16.dp)
+              .testTag("${testTagPrefix}_row"),
+      horizontalArrangement = Arrangement.Center,
+      verticalAlignment = Alignment.CenterVertically) {
+        Box(modifier = Modifier.weight(1f).testTag("${testTagPrefix}_hour")) {
+          TimeColumn(
+              selectedTime = selectedTime.hour,
+              range = 0..23,
+              onValueSelected = { onTimeSelected(selectedTime.withHour(it)) },
+              isActive = activeColumn == 0,
+              onScrollStateChanged = { isScrolling -> activeColumn = if (isScrolling) 0 else null },
+              testTag = "${testTagPrefix}_hour_picker")
+        }
+
+        Text(
+            text = ":",
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier.padding(horizontal = 4.dp),
+            color = MaterialTheme.colorScheme.onBackground)
+
+        Box(modifier = Modifier.weight(1f).testTag("${testTagPrefix}_minute")) {
+          TimeColumn(
+              selectedTime = selectedTime.minute,
+              range = 0..59 step 5,
+              onValueSelected = { onTimeSelected(selectedTime.withMinute(it)) },
+              isActive = activeColumn == 1,
+              onScrollStateChanged = { isScrolling -> activeColumn = if (isScrolling) 1 else null },
+              testTag = "${testTagPrefix}_minute_picker")
+        }
+      }
+}
+
+@Composable
+private fun TimeColumn(
+    selectedTime: Int,
+    range: IntProgression,
+    onValueSelected: (Int) -> Unit,
+    isActive: Boolean,
+    onScrollStateChanged: (Boolean) -> Unit,
+    testTag: String
+) {
+  val listState = rememberLazyListState(initialFirstVisibleItemIndex = range.indexOf(selectedTime))
```

Comment:
range is used multiple times in  here (e.g.,` range.indexOf(selectedTime)`,` range.elementAt(...)`). Since range is a IntProgression, indexOf and elementAt might not be the most incredible if range is large. Consider maybe 
Precomputing a list from the range,` like val times = range.toList()`. And use times this could improve better the performance.